resources/constraints.yaml

```diff
@@ -4,27 +4,32 @@ test:
   folds: 1
   max_runtime_seconds: 600
   cores: 4
+  max_mem_size_mb: 32768
 
-1h4c:
+1h4c32g:
```

Comment:
No, you simply can't rename all default constraints just because you need it.

As I said, the default should be `max_mem_size_mb=-1` meaning "use all memory available". That's what the existing constraint do and this must remain this way.

If you need custom constraints, please follow https://github.com/openml/automlbenchmark/blob/master/docs/HOWTO.md#constraints-definition

Reply:
I did this because your article only had configuration for benchmarks - m5.2xlarge instances, so actually 32 gb was the default value for memory constraints. Why does the number of cores have a limit in the original constraints, but the memory doesn't?

What configurations do you plan to run in the second version of paper?

Reply:
@dev-rinchin the application was originally created for research purpose and the main target to run benchmarks was AWS, and we use docker on AWS for various reasons among which: faster + more robust setup on ec2 instances, easier to maintain docker images than AWS ones, docker images can be used anywhere...

As AWS was the main target, we could rely only on instance types to specify the benchmark constraints: same number of CPUs, same amount of memory on all those instances. Later we realized that some frameworks were more greedy with disk usage and we added a constraint for this a well that allows to add additional disk space to the ec2 instances.
However, we didn't need the memory constraint because we can't add memory to the instances, and there was no good reason to artificially limit memory for example to 16GB if the instance provides 32GB.

Now these were the original requirements, and I understand that some users like you want to bring new requirements, and I'm open to support this with the following conditions:
- new requirements that are not used for the paper should be optional, and the default behaviour must behave exactly how it behaves currently. In this case, this means no artificial limitation when it's no needed.
- new requirements should be implemented consistently: if we want to have a memory constraint available for constraint definitions, it should be applied consistently for all modes (local, docker, singularity, aws...). If the goal is to support it only for docker, then it should be a docker option (declared globally in the `docker` namespace in `config.yaml`).

Also note that in default local mode and aws mode (that runs the benchmark directly in local mode on the ec2 instance), it's almost impossible to enforce this memory constraint because most frameworks simply ignore it (we would have to monitor all the processes created by the framework...).

Right now, if your goal is to run multiple docker instances on one machine, and want to limit the memory for each docker instance, I would suggest that you just close this PR and use the existing option in `config.yaml`:
```
docker:
  <<: *container
  run_extra_options: '--shm-size=1024M'
```

in your local `~/.config/automlbenchmark/config.yaml`
just define 
```
docker:
  run_extra_options: "--shm-size=1024M --memory=8g --memory-swap=16g"
```
this will be passed directly to each docker instance.
You can also pass it directly at the command line:
```
python runbenchmark.py .... -m docker -Xdocker.run_extra_options="--memory=8g --memory-swap=16g"
```

Sorry for not having thought about this option earlier.